yb-voyager/src/srcdb/srcdb.go

```diff
@@ -41,10 +41,11 @@ type SourceDB interface {
 	FilterEmptyTables(tableList []*sqlname.SourceName) ([]*sqlname.SourceName, []*sqlname.SourceName)
 	GetColumnsWithSupportedTypes(tableList []*sqlname.SourceName, useDebezium bool, isStreamingEnabled bool) (map[*sqlname.SourceName][]string, []string)
 	GetTableColumns(tableName *sqlname.SourceName) ([]string, []string, []string)
-	IsTablePartition(table *sqlname.SourceName) bool
+	ParentTableOfPartition(table *sqlname.SourceName) string
 	GetColumnToSequenceMap(tableList []*sqlname.SourceName) map[string]string
 	GetAllSequences() []string
 	GetServers() []string
+	GetChildPartitions(table *sqlname.SourceName) []*sqlname.SourceName
```

Comment:
If there are hundreds of tables that need to be exported, the "get root" and "get leaf" will be called those many number of times (which can be slow).

In future we may need to optimise this by having only one call to `GetTablePartitionTree() map[string][]string`. The implementation of the function makes only one metadata query on the source db to fetch all tables and their relationships and constructs a "tree" in memory.

Reply:
hmm yes, makes sense. will add a TODO.